odoo/fields.py

```diff
@@ -38,12 +39,29 @@
 )
 from .tools.sql import pg_varchar
 from .tools.mimetypes import guess_mimetype
-from .tools.misc import unquote, has_list_types
+from .tools.misc import unquote, has_list_types, DefaultType, DEFAULT
 from .tools.translate import html_translate, _
 
 from odoo.exceptions import CacheMiss
 from odoo.osv import expression
 
+import typing
+if typing.TYPE_CHECKING:
+    try:
+        from typing_extensions import Self
+    except ImportError:
+        from typing import Self
+    from odoo.models import BaseModel
+    from collections.abc import Mapping
+    M = typing.TypeVar("M", bound=BaseModel)
+    ModelId = int | NewId
```

Comment:
`IdType` is used only in fields.py, but declared in models.py, it is used twice to check membership. I think it should be removed:
1. Type annotations need `|` to build union types, but `isinstance` expects a tuple.
2. Both usages are `if type(value) in IdType` should become `if type(value) is int or type(value) is NewId` (note: isinstance cannot be used because we want to ignore `False`.